### ISSUE TITLE
Deserializing Get

### ISSUE THREAD

**Issue by gavin-norman-sociomantic** [sociomantic-tsunami/dhtproto#15]
It should be possible to add a function template to the received notification to deserialize structs received by Get.

**Comment by gavin-norman-sociomantic** [comment]
Something like this should do it:
```D
                static if ( ContiguousVersion.Version.Info!(T).exists )
                {
                    auto record = this.version_decorator.loadCopy!(T)
                        (serialized_record, target).ptr;
                }
                else
                {
                    Const!(void[]) cast_serialized_record = serialized_record;
                    auto record = Deserializer.deserialize(
                        cast_serialized_record, target).ptr;
                }
```

**Comment by gavin-norman-sociomantic** [comment]
Can be done in swarm, actually: https://github.com/sociomantic-tsunami/swarm/issues/105

**Comment by gavin-norman-sociomantic** [comment]
Initial proposal in swarm: https://github.com/sociomantic-tsunami/swarm/pull/106

**Comment by gavin-norman-sociomantic** [comment]
Fixed by #41.
